exporter/exporterhelper/metrics_batch.go

```diff
@@ -0,0 +1,236 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package exporterhelper // import "go.opentelemetry.io/collector/exporter/exporterhelper"
+
+import (
+	"context"
+	"errors"
+
+	"go.opentelemetry.io/collector/exporter/exporterbatcher"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+)
+
+// mergeMetrics merges two metrics requests into one.
+func mergeMetrics(_ context.Context, r1 Request, r2 Request) (Request, error) {
+	mr1, ok1 := r1.(*metricsRequest)
+	mr2, ok2 := r2.(*metricsRequest)
+	if !ok1 || !ok2 {
+		return nil, errors.New("invalid input type")
+	}
+	mr2.md.ResourceMetrics().MoveAndAppendTo(mr1.md.ResourceMetrics())
+	return mr1, nil
+}
+
+// mergeSplitMetrics splits and/or merges the metrics into multiple requests based on the MaxSizeConfig.
+func mergeSplitMetrics(_ context.Context, cfg exporterbatcher.MaxSizeConfig, r1 Request, r2 Request) ([]Request, error) {
+	var (
+		res          []Request
+		destReq      *metricsRequest
+		capacityLeft = cfg.MaxSizeItems
+	)
+	for _, req := range []Request{r1, r2} {
+		if req == nil {
+			continue
+		}
+		srcReq, ok := req.(*metricsRequest)
+		if !ok {
+			return nil, errors.New("invalid input type")
+		}
+		if srcReq.md.DataPointCount() <= capacityLeft {
+			if destReq == nil {
+				destReq = srcReq
+			} else {
+				srcReq.md.ResourceMetrics().MoveAndAppendTo(destReq.md.ResourceMetrics())
+			}
+			capacityLeft = cfg.MaxSizeItems - destReq.md.DataPointCount()
```

Comment:
Here and in the other mergesplit funcs I think it would be clearer if capacityLeft  is defined inside the for loop and then here we just do
```suggestion
			capacityLeft -= destReq.md.DataPointCount()
```

Reply:
We cannot set it inside the loop because `capacityLeft` has to be preserved between the iterations. But I can actually apply your suggestion anyway. This is the correct behavior. The current code could be considered as a bug if we had more than 2 iterations